examples/ABC_EQF.h

```diff
@@ -0,0 +1,867 @@
+#define ABC_EQF_H
+#pragma once
+
+#include <gtsam/base/Matrix.h>
+#include <gtsam/base/Vector.h>
+#include <gtsam/geometry/Rot3.h>
+#include <gtsam/geometry/Unit3.h>
+#include <gtsam/nonlinear/Values.h>
+#include <gtsam/inference/Symbol.h>
+#include <gtsam/slam/dataset.h>
+#include <gtsam/navigation/ImuBias.h>
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <cmath>
+#include <functional>
+#include <chrono>
+#include <numeric>  // For std::accumulate
+
+// All implementations are wrapped in this namespace to avoid conflicts
+namespace abc_eqf_lib {
+
+using namespace std;
+using namespace gtsam;
+
+// Global configuration
+// Define coordinate type: "EXPONENTIAL" or "NORMAL"
+extern const std::string COORDINATE;
+
+//========================================================================
+// Utility Functions
+//========================================================================
+
+
+/// Check if a vector is a unit vector
+
+bool checkNorm(const Vector3& x, double tol = 1e-3);
+
+/// Check if vector contains NaN values
+
+bool hasNaN(const Vector3& vec);
+
+/// Create a block diagonal matrix from two matrices
+
+Matrix blockDiag(const Matrix& A, const Matrix& B);
+
+/// Repeat a block matrix n times along the diagonal
+
+Matrix repBlock(const Matrix& A, int n);
+
+/// Calculate numerical differential
+
+Matrix numericalDifferential(std::function<Vector(const Vector&)> f, const Vector& x);
+
+//========================================================================
+// Core Data Types
+//========================================================================
+
+
+
+/// Input struct for the Biased Attitude System
+
+struct Input {
+    Vector3 w;               /// Angular velocity (3-vector)
+    Matrix Sigma;            /// Noise covariance (6x6 matrix)
+    static Input random();   /// Random input
+    Matrix3 W() const {      /// Return w as a skew symmetric matrix
+        return Rot3::Hat(w);
+    }
+    static Input create(const Vector3& w, const Matrix& Sigma) { /// Initialize w and Sigma
+       if (Sigma.rows() != 6 || Sigma.cols() != 6) {
+           throw std::invalid_argument("Input measurement noise covariance must be 6x6");
+       }
+       /// Check positive semi-definite
+       Eigen::SelfAdjointEigenSolver<Matrix> eigensolver(Sigma);
+       if (eigensolver.eigenvalues().minCoeff() < -1e-10) {
+           throw std::invalid_argument("Covariance matrix must be semi-positive definite");
+       }
+
+       return Input{w, Sigma};  // use brace initialization
+    }
+};
+
+/// Measurement struct
+
+struct Measurement {
+    Unit3 y;        /// Measurement direction in sensor frame
+    Unit3 d;        /// Known direction in global frame
+    Matrix3 Sigma;      /// Covariance matrix of the measurement
+    int cal_idx = -1;   /// Calibration index (-1 for calibrated sensor)
+    static Measurement create(const Unit3& y_vec, const Unit3& d_vec, /// Initialize measurement
```

Comment:
Same: kill and use brace init. Covariance checking is overkill in an example (and is not very relevant when we switch to noise models)